src/util/git.js

```diff
@@ -114,6 +115,17 @@ export default class Git implements GitRefResolvingInterface {
       };
     }
 
+    // npm local packages are specified as file:, but url parser interprets them as using the file protocol.
+    // This changes the behavior so that git doesn't see this as a hostname, but as a file path.
+    // See #3670.
+    if (parsed.protocol === FILE_PROTOCOL && !parsed.hostname && parsed.path && parsed.port === null) {
+      return {
```

Comment:
I think the only real difference between this returned object and the one below it is the `repository`.

Could this be simplified to something like 

```
function isLocalDir(parsed) {
  return parsed.protocol === FILE_PROTOCOL && !parsed.hostname && parsed.path && parsed.port === null;
}
```
```
const repository = isLocalDir(parsed) ? parsed.path : url.format({...parsed, hash: ''});
return {
  hostname: parsed.hostname || null,
  protocol: parsed.protocol || FILE_PROTOCOL,
  repository,
};
```

Reply:
Where would I go about declaring the isLocalDir function?

Reply:
I'm hitting a lot of errors in flowcheck with incompatible types trying to implement it like you suggested. Not quite sure of what I can do, as most of them point to `parsed` attributes as being undefined and incompatible with the expected types. This doesn't allow me to do a simple return in `isLocalDir()` or to assign the correct path to `repository`.

Reply:
No matter how I do it, I can't use the ternary operator because Flow doesn't really like it.

```
125:     const repository: string = isLocalDir(parsed) ? parsed.path : url.format({...parsed, hash: ''});
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ undefined. This type is incompatible with
125:     const repository: string = isLocalDir(parsed) ? parsed.path : url.format({...parsed, hash: ''});
                           ^^^^^^ string
```

Of course, it doesn't complain if repository isn't defined as `string`, but then I can't use it inside the return value because the `GitUrl` type specifies `repository` as `string`. I'm not sure how you'd want to do it because without the ternary operator, it's not as simple as a single `const repository` and then we're back to separate checks.

Reply:
Oh yeah, Flow likes to yell at me too 😆 

I'm guessing that is because `parsed.path` could be `undefined`. I was just trying to simplify the code a bit, but if Flow is going to fight you, then don't worry about it.